flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/DebeziumSourceFunction.java

```diff
@@ -345,73 +376,60 @@ public void run(SourceContext<T> sourceContext) throws Exception {
                 properties.getProperty(
                         Heartbeat.HEARTBEAT_TOPICS_PREFIX.name(),
                         Heartbeat.HEARTBEAT_TOPICS_PREFIX.defaultValueAsString());
-        this.debeziumConsumer =
-                new DebeziumChangeConsumer<>(
+        this.debeziumChangeFetcher =
+                new DebeziumChangeFetcher<>(
                         sourceContext,
                         deserializer,
                         restoredOffsetState == null, // DB snapshot phase if restore state is null
-                        this::reportError,
-                        dbzHeartbeatPrefix);
+                        dbzHeartbeatPrefix,
+                        handover);
 
         // create the engine with this configuration ...
         this.engine =
                 DebeziumEngine.create(Connect.class)
                         .using(properties)
-                        .notifying(debeziumConsumer)
+                        .notifying(changeConsumer)
                         .using(OffsetCommitPolicy.always())
                         .using(
                                 (success, message, error) -> {
-                                    if (!success && error != null) {
-                                        this.reportError(error);
+                                    debeziumChangeFetcher.close();
```

Comment:
Don't need to close fetcher here, just need to report error and the fetcher will be closed .